guidelines/application_requirements.md

```diff
@@ -0,0 +1,84 @@
+# Guidelines for Application Requirements
+
+Authors: chasm@google.com, jlewi@google.com, kam.d.kasravi@intel.com
+
+## Objective
+
+Define requirements and recommendations aimed at graduating applications (e.g. controller and web apps) in Kubeflow to 1.0.
+
+## Scope
+
+This document is intended to cover applications that are deployed in the Kubernetes cluster and in particular:
+
+1. Custom resources and controllers
+1. Web applications
+
+This document is not intended to cover SDKs (e.g. fairing) or client side tools (like kfctl).
+
+## Goals
+
+The purpose of this doc is to:
+
+1. Establish criterion for graduating a Kubeflow application to a supported Kubeflow application; i.e. 1.0
+1. Set clear expectations for users regarding Kubeflow applications that are 1.0
+
+## Requirements
+
+### Configuration and deployment
+
+| Description | Category | Explanation |
+|-------------|----------|-------------|
+| Kustomize package  | Required  | <ul><li>Kubeflow has standardized on Kustomize for configuring and deploying packages </ul> |
+| Application CR  | Required  | <ul><li>Kubeflow has standardized on using the Kubernetes Application CR to provide consistent metadata and monitoring about applications <li> Application CR should be an owner of all resources so that deleting the application CR uninstalls the application </ul> |
+| ```app.kubernetes.io``` labels on every resource | Required  | <ul><li>Every resource that is part of the application should include the [labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels) recommended by Kubernetes </ul> |
+| Images listed in kustomization.yaml | Required  | <ul><li>All docker images used by the application should be listed in one or more kustomization.yaml files <li>This facilitates mirroring the images to different docker registries </ul> |
+| Upgradeability | Required  | <ul><li>Applications must support upgrading between consecutive major and minor releases </ul> |
+| Separate cluster scoped and namespace scoped resources | Recommended  | <ul><li>To the extent possible cluster scoped resources should be installable separately (e.g. via a separate kustomize package) <li> This allows cluster admins to install only the cluster scoped resources </ul> |
```

Comment:
It would be also better to document cluster scoped vs namespace scoped behavior that has to be supported in each application.  eg: KFServing operator, Katib operator don't support namespace scope while Tfjob/PyTorch Job supports namespace scope in its implementation.

Reply:
Apologies @richarddli I meant @richardsliu